x/observer/keeper/msg_server_add_observer.go

```diff
@@ -52,10 +52,15 @@ func (k msgServer) AddObserver(
 		return &types.MsgAddObserverResponse{}, nil
 	}
 
-	k.AddObserverToSet(ctx, msg.ObserverAddress)
+	// Add observer to the observer set
+	err = k.AddObserverToSet(ctx, msg.ObserverAddress)
+	if err != nil {
+		return &types.MsgAddObserverResponse{}, err
+	}
 	observerSet, _ := k.GetObserverSet(ctx)
```

Comment:
We should think this, if not found we should explicitely set ObserverSet empty
```
if !found {
  observerSet = types.ObserverSet{}
}
```

Reply:
Not sure what you mean, if not found we set it it with the address provided 
https://github.com/zeta-chain/zeta-node/blob/d4048a53b75a178d8519057443d1c85afddc08ba/x/observer/keeper/observer_set.go#L44-L47

Do you mean , for the case !found , and address is not valid ?

Reply:
My point is if `found == false`, `observerSet` should be deemed as undefined and not used